tests/random_lax_test.py

```diff
@@ -365,6 +366,47 @@ def testCategorical(self, p, axis, dtype, sample_shape):
         pmf = lambda x: np.where(x < len(p), p[np.minimum(len(p) - 1, x)], 0.0)
         self._CheckChiSquared(samples, pmf=pmf)
 
+  @jtu.sample_product(
+    [
+      {
+        "logits_shape": logits_shape,
+        "axis": axis,
+      }
+      for logits_shape in [(2,), (3,), (2, 3), (2, 5), (2, 3, 5)]
+      for axis in range(-len(logits_shape), len(logits_shape))
+    ],
+    batch_shape=[(), (2,), (3,), (2, 3), (2, 5), (2, 3, 5), None],
```

Comment:
Maybe cleaner to leave out `None` here, as `()` covers that case – then you could remove the `if` block below.